packages/pilot/src/pages/Balance/Balance.js

```diff
@@ -322,33 +406,62 @@ class Balance extends Component {
       && company.default_recipient_id
     ) {
       const pathId = getValidId(getRecipientId(company), null)
-      this.updateQuery(this.state.query, pathId)
+
+      return this.updateQuery(this.getQuery(), pathId)
     }
+
+    return undefined
+  }
+
+  getQuery (newTimeframe) {
+    const {
+      history: {
+        location: {
+          search,
+        },
+      },
+    } = this.props
+    const timeframe = getTimeframeProp(qs.parse(search))
+    const { query } = this.state
+
+    return query[newTimeframe || timeframe]
+  }
+
+  setQuery (query) {
+    const { query: stateQuery } = this.state
+    const timeframe = getTimeframeProp(query)
+    const newQuery = assoc(timeframe, query, stateQuery)
+    this.setState({
+      query: newQuery,
+    })
   }
 
   updateQuery (query, id) {
+    const timeframe = getTimeframeProp(query)
     const buildBalanceQuery = pipe(
       queryDatesToISOString,
       merge(query),
       qs.stringify
     )
-
-    const queryObject = isNilOrEmpty(query)
-      ? this.state.stateQuery
-      : query
-    const pathId = getValidId(getRecipientId(this.props.company), id)
-
     const {
       history,
       match: {
         params,
       },
     } = this.props
 
+    const queryObject = isNilOrEmpty(query)
+      ? this.getQuery()
+      : query[timeframe] || query
```

Comment:
poderia criar uma função nessa parte aqui pra não ficar tão estranho a leitura
```suggestion
      : query[timeframe] || query
```